app/services/csv_exporter.rb

```diff
@@ -4,18 +4,15 @@ module CsvExporter
   def self.export(resources, columns, header = nil)
     header ||= default_header(columns)
     raise ArgumentError, "Columns and header row aren't the same length" unless columns.length == header.length
-    # need to save the current context as the enumerator is executed by a separate thread
-    context = Foreman::ThreadSession::Context.get
-    Enumerator.new do |csv|
-      Foreman::ThreadSession::Context.set(context)
-      csv << CSV.generate_line(header)
-      columns.map!{|c| c.to_s.split('.').map(&:to_sym)}
-      resources.uncached do
-        resources.reorder(nil).find_each do |obj|
-          csv << CSV.generate_line(columns.map{|c| c.inject(obj, :try)})
-        end
+    csv = []
+    csv << CSV.generate_line(header)
+    columns.map!{|c| c.to_s.split('.').map(&:to_sym)}
```

Comment:
@tbrisker mind having a look?

Reply:
From a quick look this doesn't seem like a good solution. The reason this was done using an enumerator is to allow streaming the response instead of building it in memory and sending only after it completes- this can be a file several GB large that takes a few minutes to build completely. Not sure what is the reason for this change but we have to keep this as a stream.

Reply:
The reason was that setting context in test the way it was done no longer worked.